launcher/src/main/java/org/apache/spark/launcher/SparkLauncher.java

```diff
@@ -107,6 +121,30 @@ public static void setConfig(String name, String value) {
     launcherConfig.put(name, value);
   }
 
+  /**
+   * Specifies that Spark Application be stopped if current process goes away.
+   * It tries stop/kill Spark Application if {@link LauncherServer} goes away.
+   *
+   * @since 2.2.0
+   * @return This launcher.
+   */
+  public SparkLauncher stopIfLauncherShutdown() {
```

Comment:
Same here; `stopOnShutdown` more closely matches the behavior, since the behavior is not based on the launcher, but the current app shutting down.

Also, does this work at all for child processes? I don't think so. So the javadocs should at least say that.

Following from my previous comments, I'm not even sure this option is necessary. When would you not want this when using child threads? The user can call `disconnect()` if he doesn't want the child app to fate-share with the launcher. That would simplify the API.